packages/core/admin/admin/src/hooks/useFetchPermissionsLayout/index.js

```diff
@@ -1,30 +1,39 @@
 import { useEffect, useReducer } from 'react';
-import { request } from '@strapi/helper-plugin';
+import { useFetchClient } from '@strapi/helper-plugin';
 
 import reducer, { initialState } from './reducer';
 
+/**
+ * TODO: refactor this to use react-query and move it to the `Roles` SettingsPage
+ */
 const useFetchPermissionsLayout = (id) => {
   const [{ data, error, isLoading }, dispatch] = useReducer(reducer, initialState);
+  const { get } = useFetchClient();
 
   useEffect(() => {
     const getData = async () => {
-      dispatch({
-        type: 'GET_DATA',
-      });
-
-      const { data } = await request('/admin/permissions', {
-        method: 'GET',
-        params: { role: id },
-      });
-
-      dispatch({
-        type: 'GET_DATA_SUCCEEDED',
-        data,
-      });
+      try {
+        dispatch({
+          type: 'GET_DATA',
+        });
+
+        const {
+          data: { data },
+        } = await get('/admin/permissions', {
+          params: { role: id },
+        });
+
+        dispatch({
+          type: 'GET_DATA_SUCCEEDED',
+          data,
+        });
+      } catch (err) {
+        // silence is golden
```

Comment:
😸

Reply:
haha I should explain, originally there was no handling for this so if the request was cancelled e.g. by the component un-mounting then the whole app crashes, this stops the app from crashing but behaves exactly as it did before 👀